### DISCUSSION TITLE
What happened to Rect.Empty, Rect.IsEmpty and Size.Empty?

### DISCUSSION THREAD

**Discussion by kuiperzone** [AvaloniaUI/Avalonia#10868] [General]
In 11P6 the following have been removed:

`Rect.Empty`
`Rect.IsEmpty`
`Size.Empty`

and I guess:

`Size.IsEmpty`

My question is why remove these?

* They were useful
* They are in WPF
* They were absolutely trivial to implement

Yes, I can code around their absence. But I made extensive use of them and only see their removal as another breaking change.

**Comment by kekekeks** [comment]
@robloo @maxkatz6

**Comment by robloo** [comment]
@kuiperzone Please read the latest discussions here:

 * https://github.com/AvaloniaUI/Avalonia/issues/10264
 * https://github.com/AvaloniaUI/Avalonia/issues/7762 (original)

For you comments specifically:

 1. Rect is not as trivial as you think. There is a lot more to this discussion and Avalonia was not actually written following WPF.
 2. Size should probably be Zero/IsZero instead now
 3. In most cases comparison to "empty" can be replaced with "default(Type)"

I would suggest just making some extension methods in your project implementing these how you want it for now.

Edited to add num.3 which is an important point I missed but was reminded of below.

**Comment by maxkatz6** [comment]
To add to the @robloo's message. In most cases they could be replaced with "default(Type)".

**Comment by kuiperzone** [comment]
OK. Thanks.
